Generator/Sources/NeedleFramework/Generating/Serializers/OutputSerializer.swift

```diff
@@ -37,23 +37,74 @@ class OutputSerializer: Serializer {
     ///
     /// - returns: The source code `String`.
     func serialize() -> String {
-        let registrationBody = providers
-            .map { (provider: SerializedProvider) in
-                provider.registration
-            }
-            .joined()
-            .replacingOccurrences(of: "\n", with: "\n    ")
+        // Dependency factories have deterministic names, so we can only define a
+        // factory once, or the code won't compile.
+        var registeredFactories: Set<String> = []
```

Comment:
minor nit: Since you check for the existence before adding this could just be an array.

Reply:
I was using a `Set` so that the check would be O(1) instead of O(log n) (assuming we sorted, or O(n) otherwise).